### ISSUE TITLE
Complex intervals

### ISSUE THREAD

**Issue by dpsanders** [dpsanders/ValidatedNumerics.jl#90]
It is currently possible to create a complex interval with e.g.

```julia
julia> a = 3 + @interval(4)*im
[3.0, 3.0] + [4.0, 4.0]*im
```

We should be able to write `@interval(3+4im)` to get this.

EDIT: We then need to see how to define things like `exp(a)`

**Comment by dpsanders** [comment]
For `exp(a)` it turns out that we just need to define

```
Base.isfinite(x::Interval) = isfinite(x.lo) && isfinite(x.hi)
```
and then it works.

**Comment by lbenet** [comment]
I think this should not be too difficult, and somewhere I read that it is of interest. The way to go is allow the parametric constructor to allow act over `Number`s: `T<:Number`, and probably make `Interval` also a subtype of `Number`.

**Comment by dpsanders** [comment]
Since `im` is of the strange type `Complex{Bool}`, something like the following works:
```julia
ValidatedNumerics.make_interval(Float64, x::Complex{Bool}) = (x == im) && one(Float64)*im
```

**Comment by dpsanders** [comment]
Implemented in #98.
